test/model/host_report_status_test.rb

```diff
@@ -0,0 +1,173 @@
+      assert_equal HostStatus::Global::WARN, status.to_global
+    end
+  end
+
+  test 'status is disabled when reporting is disabled' do
+    host.enabled = false
+    assert_equal HostStatus::HostReportStatus::DISABLED, status.refresh
+  end
+
+  test '#out_of_sync? is false when reported_at is unknown' do
+    status.reported_at = nil
+    refute status.out_of_sync?
+  end
+
+  test '#out_of_sync? is false when window is big enough' do
+    original, Setting[:outofsync_interval] = Setting[:outofsync_interval], (Time.now.utc - report.reported_at).to_i / 60 + 1
+    refute status.out_of_sync?
+    Setting[:outofsync_interval] = original
+  end
+
+  describe '#out_of_sync?' do
+    let(:report) { FactoryBot.create(:host_report, reported_at: Time.now.utc - 1.year) }
+
+    test '#out_of_sync? is false when out of sync is disabled' do
+      status.stubs(:out_of_sync_disabled?).returns(true)
+      refute status.out_of_sync?
+    end
+
+    context 'with last report format' do
+      setup do
+        status.last_report.stubs(:format).returns('Test')
+      end
+
+      test 'is false when formats out of sync is disabled' do
+        stub_outofsync_setting(true)
+        refute status.out_of_sync?
+      end
+
+      test "is true when formats out of sync isn't disbled and it is ouf of sync" do
+        stub_outofsync_setting(false)
+        status.reported_at = '2015-01-01 00:00:00'
+        status.save
+        assert status.out_of_sync?
+      end
+
+      def stub_outofsync_setting(value)
+        Foreman.settings._add('test_out_of_sync_disabled',
+          context: :test,
+          type: :boolean,
+          category: 'Setting',
+          full_name: 'Test out of sync',
+          description: 'description',
+          default: false)
+        Setting[:test_out_of_sync_disabled] = value
+      end
+    end
+  end
+
+  test '#refresh! refreshes the date and persists the record' do
+    status.expects(:refresh)
+    status.refresh!
+
+    assert status.persisted?
+  end
+
+  test '#refresh updates date to reported_at of last report' do
+    status.reported_at = nil
+    status.refresh
+
+    assert_equal report&.reported_at&.to_i, status&.reported_at&.to_i
+  end
+
+  test '#relevant? only for hosts with #configuration? true, or a last report, or setting enabled' do
+    host.expects(:configuration?).returns(true)
+    assert status.relevant?
+
+    host.expects(:configuration?).returns(false)
+    status.expects(:last_report).returns(mock)
+    assert status.relevant?
+
+    host.expects(:configuration?).returns(false)
+    status.expects(:last_report).returns(nil)
+    refute status.relevant?
+
+    host.expects(:configuration?).returns(false)
+    status.expects(:last_report).returns(nil)
+    Setting[:always_show_configuration_status] = true
+    assert status.relevant?
+  end
+
+  test 'overwrite outofsync_interval as host parameter' do
+    window = 30
+    Setting['puppet_interval'] = 10
```

Comment:
This yells `Setting definition for 'puppet_interval' not found, can not set`.

Reply:
@lzap I have this error as well